src/xcode/ENA/ENA/Source/HTTPClientCore/Security/EvaluateTrust/DefaultTrustEvaluation.swift

```diff
@@ -35,12 +40,10 @@ class DefaultTrustEvaluation: TrustEvaluating {
 		}
 #endif
 		
-		// we expect a chain of at least 2 certificates
-		// index '1' is the required intermediate
-		guard let serverCertificate = SecTrustGetCertificateAtIndex(trust, 1),
+		guard let serverCertificate = SecTrustGetCertificateAtIndex(trust, certificatePosition),
 			  let serverPublicKey = SecCertificateCopyKey(serverCertificate),
 			  let serverPublicKeyData = SecKeyCopyExternalRepresentation(serverPublicKey, nil ) as Data?,
-			  publicKeyHash == serverPublicKeyData.sha256String()
+			  publicKeyHash == serverPublicKeyData.sha256()
 		else {
 			trustEvaluationError = .default(.CERT_MISMATCH)
 			completionHandler(.cancelAuthenticationChallenge, /* credential */ nil)
```

Comment:
This means we still will show the "no internet" error if the trust eval will fail?

Reply:
No we show the error which is inside of trustEvaluationError (CERT_MISMATCH).
The trustEvaluationError was not working before, fallback was then "no internet".